### ISSUE TITLE
Why numbers as identifiers?

### ISSUE THREAD

**Issue by mk-pmb** [stevemao/better-than-before#6]
I'd think that strings with good names would be way easier for humans to remember and manage.
In cases where you don't need fancy list selection ("run all setups that match this regexp except these") you won't even need to store the setup function on a separate object, you could just use named function statements.

**Comment by stevemao** [comment]
Can you provide a sample code snippet how you want the api look like?

**Comment by mk-pmb** [comment]
Nope, unfortunately I can't. I tried three times and each time I ended up with the example no longer using this module. =) Even when I assume some magic about implicit number ranges, an array slice seems easier to me.
Could you explain the advantage of calling functions by number over calling regular named functions?
Maybe I can then use that and show how it would be even better with names.

**Comment by stevemao** [comment]
I don't really understand your question. Are you talking about `preparing(2); // I need both first and second setup`? It just means I only want the first 2 setups. If the third one is run before it would be cleaned before running the first 2 setups.

**Comment by mk-pmb** [comment]
So it seems to be about cleaning up those setups that are no longer required? Then I'd prefer a dependency system, so `beforeEach` can lookup which amenities my test needs, prepare the missing ones and clean up left-over amenities that the current test doesn't need. A deluxe version would allow amenities to define their own dependencies so you'd get a dependency graph, but for a start, a list of required setups might do. It could look like this:
Edit: [made it a gist.](https://gist.github.com/mk-pmb/c8f4918d101d78ffe6dc48f330bbbf6c)
